backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,151 @@
+import java.util.Date;
+import java.util.Calendar;
+import java.util.Objects;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+
+/** */
+@RunWith(JUnit4.class)
+public final class FindMatchQueryTest {
+
+  // Some people that we can use in our tests
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+  private static final String PERSON_C = "Person C";
+  private static final String PERSON_D = "Person D";
+
+  // Default parameters unused in query
+  private static final long ID_DEFAULT = 0;
+  private static final String TIMEZONE_DEFAULT = "";
+  private static final long TIMESTAMP_DEFAULT = 0;
+
+  // Some times available until on 6/25/2020
+  private static final long TIME_0250PM = Time.getTimeMillis(14, 50);
+  private static final long TIME_0330PM = Time.getTimeMillis(15, 30);
+  private static final long TIME_0400PM = Time.getTimeMillis(16, 0);
+  private static final long TIME_0600PM = Time.getTimeMillis(18, 0);
+
+  private static final int DURATION_15_MINUTES = 15;
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_45_MINUTES = 45;
+  private static final int DURATION_60_MINUTES = 60;
+
+  private FindMatchQuery query;
+  private Time time;
+
+  @Before
+  public void setUp() {
+    // Set "current" date to  1/1/2020 2:00pm GMT
+    Calendar c = Calendar.getInstance();
+    c.set(2020, 0, 1, 14, 0, 0);
+    c.set(Calendar.MILLISECOND, 0);
+    Date date = c.getTime();
+
+    query = new FindMatchQuery(date);
+    time = new Time(date);
+  }
+
+  @Test
+  public void compatibleTimeAndDuration() {
+    /* Two participants who are compatible in available time AND duration */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB));
+
+    assertThat(actual.getFirstParticipant().getLdap().equals(PERSON_B));
+    assertThat(actual.getSecondParticipant().getLdap().equals(PERSON_A));
+    assertThat(actual.getDuration() == DURATION_15_MINUTES);
+  }
+
+  @Test
+  public void compatibleTime() {
+    /* Two participants who are compatible in available time but NOT duration */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB));
+
+    assertThat(Objects.isNull(actual));
```

Comment:
assertThat(actual).isNull()

As written this assertion is only partially complete and so the test will pass even if `actual` isn't null. To assert on booleans you would write

`assertThat(myBoolean).isTrue()` or `assertThat(myBoolean).isFalse()`

Reply:
Got it.